libs/vkd3d/command.c

```diff
@@ -2725,6 +2728,16 @@ static void d3d12_command_list_update_packed_descriptors(struct d3d12_command_li
                 root_signature->packed_descriptor_set,
                 1, &descriptor_set, 0, NULL));
     }
+
+    /* Set descriptor offsets */
+    if (root_signature->descriptor_table_count)
```

Comment:
This is called only inside update_packed_descriptors. This seems a bit sketchy if we only changed a root descriptor which uses descriptor tables. I think we technically don't need to update packed descriptors in that case, only push constants.

Reply:
True, on the other hand we currently have no way to check whether a table has packed descriptors or not.

Reply:
If any SetDescriptorTable call will invalidate the packed set, I suppose this is fine, but I'd like a TODO comment explaining what optimization we can do later so we don't forget we need to move this out.

Reply:
Added a comment.

I guess for an implementation, we could rework this further and not track each individual table, instead just adding a `packed_descriptors_dirty` and `descriptor_tables_dirty` flag similar to the static samplers one, and store whether a table has packed descriptors with the root signature.

Reply:
ye, I think that'd be a good thing to do since it's unlikely UAV/CBV ranges will overlap with SRVs for bindless scenarios. But it doesn't block this PR.